lib/cocoapods/sandbox/headers_store.rb

```diff
@@ -24,6 +24,7 @@ def initialize(sandbox, relative_path)
         @sandbox       = sandbox
         @relative_path = relative_path
         @search_paths  = []
+        @cache = {}
```

Comment:
Recursively searching for header search targets, I figured a cache will help `header_store`. It actually does as I saw almost minimal increase in `pod install` times, versus without the cache of 5-10 seconds.